### ISSUE TITLE
Feature request: blank (absolute) style

### ISSUE THREAD

**Issue by robinjoha** [dsifford/academic-bloggers-toolkit#155]
Hi. I have used different styles, but sometimes I just want a complete blank style. Really blank. A textform to fill out, with formatting options (bold/italic/underline/url etc) and nothing more. There should be a choice on how the inline citation should be, like numbers etc. But other than that, just complete blank.

It would make my day for sure. Then you could also use this for "notes" other than just references.

**Comment by robinjoha** [comment]
For examples something like this:

1. I visited **this town** last year, _but also in 2001_ - see more about this town here: [something.com/some-url](something.com/some-url)
2. See also my earlier posts about this topic: [something.com/some-other-url](something.com/some-other-url) 
3. I have written about this topic before, so take a look through my archives to see other examples.

**Comment by fbennett** [comment]
For flexibility, maybe this could be addressed by auxiliary plugins to support references stored in Zotero and Mendeley? Both tools have APIs that can provide references in the format used in ABT. A user managing personal references in one of those could then easily reuse them across multiple documents.

**Comment by dsifford** [comment]
@fbennett which format would those be? `Generic`? I could probably just build that into the plugin with the other ones.. (I skipped a whole bunch of the rarely-used ones).

**Comment by fbennett** [comment]
Actually, I think misunderstood the gist of the feature request. I'll have
to read through the past posts (as I should have done in the first place).

**Comment by fbennett** [comment]
Ha. There aren't earlier posts, those links are just examples, aren't they.

Those "blank references" aren't references as the processor understands
them. You could get something like this with a "note" style, if the editor
has the ability to set footnotes/endnotes independently of the citation
processor.

**Comment by dsifford** [comment]
@fbennett Haha no worries! I thought that might be the case. 

@robinjoha I'll have to think this over a bit to see if this will be something I could reasonably add to the plugin. Although it sounds like a benign addition, it will undoubtedly add a large layer of complexity for the processor(s) to be able to reason through them while maintaining the document's bibliographic citations in the correct order.

So, in short.... Maybe! I'll keep you posted.

**Comment by robinjoha** [comment]
It could also be a "Citation Type" instead of a style - just like Journal Article, Book and Web Page for instance. Just call it "Blank" or something like that.

And one text-field only - with (or without) formatting options. No need for author, title etc. Just "content" and that is all. And/if more options, just make them all not mandatory.

@fbennett the links where just examples of references 

Also

@dsifford also, how can you actually edit a reference already existing? I can't figure it out. I have to delete it and then reinsert. And the last time, even though I added the correct one, it just pasted the earlier (now deleted) reference?

**Comment by dsifford** [comment]
> It could also be a "Citation Type" instead of a style - just like Journal Article, Book and Web Page for instance. Just call it "Blank" or something like that.

This makes sense in theory, however the logic behind how the citations are physically generated is incredibly complex. In order for the processor to be agnostic to any one specific citation style, it must abide by a [strict specification](http://docs.citationstyles.org/en/stable/specification.html). Included in this spec is a set of [citation types](http://docs.citationstyles.org/en/stable/specification.html#appendix-iii-types) that are available to choose from. 

@fbennett will be able to answer this question better than I can, but there may be a _slight_ chance that you might be able to get away with inserting a "Generic" citation type with just a title and nothing else. However, because most citation styles will append `n.d.`, `n.p.`, etc. to the fields that are left blank, it's likely that the only way you'd be able to get away with what you're trying to accomplish is by [defining and using your own custom style](https://github.com/dsifford/academic-bloggers-toolkit/wiki#using-a-custom-citation-style) specifically purposed for your use-case.

> how can you actually edit a reference already existing?

[This](https://github.com/dsifford/academic-bloggers-toolkit/wiki#adding-additional-references-to-existing-citation-blocks) might be helpful for what you're describing.

Here it is in action:

![image](http://i.giphy.com/26uf6y7SHkFqT3eCY.gif)

 I might be misunderstanding your question though. If so, can you show me a screen recording (best) or screenshot (acceptable) of what you're describing?

Thanks

**Comment by fbennett** [comment]
Unfortunately, having everything in one field would break in styles that
sort the bibliography according to field-specific rules. (Also, the item
types available are specified in CSL, so it's not possible to add fresh
ones without going through specification review.)

**Comment by dsifford** [comment]
@fbennett while we're on this topic, is there a list somewhere that you know of that states the required fields for each of the defined citation types? That was one thing I struggled with when initially putting this together (and was one of the reasons that I didn't add all of the citation types in the first place)

**Comment by dsifford** [comment]
@robinjoha New developments have been made.. Might be able to add something along the lines of what you're requesting.

Standby for a demo....

**Comment by dsifford** [comment]
@robinjoha Let me know if this is similar to what you were looking for....

![gif](http://i.giphy.com/d31w2OMZyQrVFV4Y.gif)

I haven't tested this heavily, but I think it may work.

**Comment by robinjoha** [comment]
Yeah! That was what I was requesting, I would have liked to have formatting options as well, but that is not as important. Very good!

**Comment by dsifford** [comment]
Links will automatically be converted. Can't add rich formatting because that would make the processor fragile + would also allow people to circumvent style rules.

**Comment by dsifford** [comment]
@robinjoha 

I tested this on a handful of different citation types sorted by a handful of different sort-keys.

If the citation type specifies a sort-key that isn't available in the generic "note", it'll spit out an error in the document. 

![image](https://cloud.githubusercontent.com/assets/5240018/18563421/92aabd06-7b55-11e6-8c9b-6770564ad86f.png)

Because this isn't a "crash" and the plugin seems to still operate smoothly, I'll be able to add this in. Hopefully you aren't using one of the styles that have unusual sort-keys. 

@fbennett thoughts on this? Seems reasonable?

**Comment by robinjoha** [comment]
Looks great in my opinion.

**Comment by dsifford** [comment]
@robinjoha Just pushed out the next release that adds this feature. 

Let me know if it works out for you. 👍

**Comment by robinjoha** [comment]
Brilliant! Just what I needed!
